Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -275,8 +289,8 @@ using namespace amrex;
                         // Mirror location inside the domain by "ig" number of cells
                         // across PEC boundary in direction, idim, and side, iside
                         ijk_mirror[idim] = ( (iside == 0)
-                                       ? (dom_lo[idim] + ig)
-                                       : (dom_hi[idim] + is_nodal[idim] - ig));
+                                        ? (dom_lo[idim] + ig - (1 - is_nodal[idim]))
+                                        : (dom_hi[idim] + 1 - ig));
```

Comment:
Why does the second line (the high side) not take into account the staggering along `idim` (here and in the similar code lines above too)?

I would have expected something like the following, if we are mirroring _inside_ the domain (which seems to be the case, since we are adding, not subtracting, `ig` in the low side case, but I may be missing something):
```suggestion
                                        : (dom_hi[idim] + (1 - is_nodal[idim]) - ig));
```
I'm thinking of situations like the following cases (and thinking of a case where `ig=1`):
```
low side nodal:
--o-----[-----o--
        0

low side staggered:
--|--x--[--x--|--
           0

high side nodal:
--o-----]-----o--
        n

high side staggered:
--|--x--]--x--|--
    n-1
```

Reply:
Thanks for checking and thinking about these indices. I think you are right that there is still a bug here. The asymmetry between the lo and hi sides incorporating the nodal flag in the highlighted code comes from the fact that the nodal flag is used when calculating `ig` for the hi case but not for the low case. If you plug the calculations for `ig` into the calculation of the mirror cells you get:

lo case: `ijk_mirror = dom_lo - (1 - is_nodal) - (dom_lo - ijk_vec)`
hi case: 'ijk_mirror = dom_hi + (1 + is_nodal) - (ijk_vec - dom_hi)`

which shows the current treatment of the two cases more clearly. I think the fix is to change the calculation of `ig` in the hi case to just be `ig = ijk_vec[idim] - dom_hi[idim]` and add your suggested change. I'll plot the field values to confirm the fix.

Reply:
Looking at it again, actually for nodal fields the high side boundary is at cell `dom_hi + 1` (so n+1 in your diagram), that is why we need to compare to   `dom_hi + is_nodal`.

Reply:
So basically as is, the high side mirror calculation can be written as:
`ijk_mirror = (dom_hi + is_nodal) + (1 - is_nodal) - [ijk_vec - (dom_hi + is_nodal)]`
which I believe is correct.

Reply:
@roelof-groenewald, @EZoni is this fixed? :)

Reply:
I think the change for the hi-condition should be 
`dom_hi[idim] + is_nodal[idim] - ig` 
This way, if its nodal, we add 1, else 0 
not `1-is_nodal[idim]` 
@EZoni do you agree?

Reply:
I agree with @roelof-groenewald 
changing the hi condition to `ijk_mirror = (dom_hi + is_nodal) + (1 - is_nodal) - ig`
is general and correct for nodal and cell-centered staggering. 
Thank you @roelof-groenewald !